src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,83 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+    private int currentSize;
+    private Node<K, V>[] node;
+
+    public MyHashMap() {
+        node = new Node[DEFAULT_INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        resize();
+        Node<K, V> currentNode = node[getIndex(key)];
+        if (currentNode == null) {
+            node[getIndex(key)] = new Node<>(key, value);
+            currentSize++;
+            return;
+        }
+        while (currentNode.next != null || Objects.equals(currentNode.key, key)) {
+            if (Objects.equals(currentNode.key, key)) {
+                currentNode.value = value;
+                return;
+            }
+            currentNode = currentNode.next;
+        }
+        currentNode.next = new Node<>(key, value);
+        currentSize++;
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        Node<K, V> currentNode = node[getIndex(key)];
+        if (currentNode == null) {
+            return null;
+        }
+        while (!Objects.equals(currentNode.key, key)) {
+            currentNode = currentNode.next;
+        }
+        return currentNode.value;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return currentSize;
+    }
+
+    private void resize() {
+        if (currentSize > node.length * LOAD_FACTOR) {
+            currentSize = node.length * 2;
+            Node<K, V>[] oldNode = new Node[currentSize];
+            Node<K, V>[] newNode = node;
+            node = oldNode;
+            currentSize = 0;
```

Comment:
```suggestion
            int newCapacity = node.length * 2;
            Node<K, V>[] oldNodes = new Node[currentSize];
            Node<K, V>[] newNodes = node;
            node = oldNode;
            currentSize = 0;
```
it's not obvious why you change currentSize if this value is responsible for storing number of elements in the HashMap

Reply:
so not to create an additional variable or it doesn't make a lot of sense in methods?

Reply:
if you're talking about 'newCapacity' it's just not so obvious why it has been reseted to null if you don't remove all the elements, that's why I proposed naming that will be more suited